### PR TITLE
Added IntegrationFunction functions to add a source function node to …

### PR DESCRIPTION
**Pull request by jbaileyhandle**
…the integration function, combine two nodes with a mux, and fetch integration function operands for a given node.

### CODE REVIEW

**Review by meheff — CHANGES_REQUESTED** [review]
Just some nits.

File: xls/contrib/integrator/ir_integrator.h [hunk 1]
```diff
@@ -43,6 +43,18 @@ class IntegrationFunction {
 
   Function* function() const { return function_.get(); }
 
+  // Add the external node 'to_insert' into the integration function. Mapped
+  // operands are automatically discovered and connected.
+  absl::StatusOr<Node*> InsertNode(const Node* to_insert);
+
+  // For the integration function nodes node_a and node_b,
+  // returns a single integration function node that combines the two
+  // nodes. This may involve adding a mux and parameter select signal.
```
[the remark this answers is not in the document]
Comment (jbaileyhandle) [reply xls/contrib/integrator/ir_integrator.h]: Nope.  What I meant was we might have to add a new parameter which will be used as the select signal for the new mux.  Clarified comment.

File: xls/contrib/integrator/ir_integrator.h [hunk 2]
```diff
@@ -43,6 +43,18 @@ class IntegrationFunction {
...
+  // If provided, the bool pointed to  by 'new_mux_added' will be be
+  // set to true if this call added a new mux.  Otherwise, false.
+  absl::StatusOr<Node*> UnifyIntegrationNodes(Node* node_a, Node* node_b,
+                                              bool* new_mux_added = nullptr);
```
[the remark this answers is not in the document]
Comment (jbaileyhandle) [reply xls/contrib/integrator/ir_integrator.h]: Done.

File: xls/contrib/integrator/ir_integrator.cc [hunk 3]
```diff
@@ -111,6 +111,77 @@ IntegrationFunction::GetNodesMappedToNode(const Node* map_target) const {
...
+                            node->operands().end());
+    return operand_mappings;
+  }
+
+  for (const auto* operand : node->operands()) {
+    if (!HasMapping(operand)) {
+      // TODO(jbaileyhandle): Implement.
+      return absl::UnimplementedError(
+          "GetIntegratedOperands for unmapped operands not yet implemented");
+    } else {
```
[the remark this answers is not in the document]
Comment (jbaileyhandle) [reply xls/contrib/integrator/ir_integrator.cc]: Done.

**Pull request merged**